duckduckgo/Android
# Feature/craig/lint checking

Task/Issue URL: https://app.asana.com/0/488551667048375/1176263457388604
Tech Design URL: 
CC: 

**Description**:
Adds ability to automate code formatting checks using `ktlint`.

**Steps to test this PR**:
1. Install `develop` first and run the app. Then install this version to ensure no migration issues from the change (a DB entity class was renamed because it violated the code style)
1. Run `./gradlew ktlint` and verify there are no reported violations.
1. Deliberately add a violation to a Kotlin class (e.g., add a bunch of useless new lines) and run `ktlint` again. This time, verify the build fails and reports the issue.

ℹ️ Note, the Bitrise checks **do not** yet enforce these rules. There is a new workflow ready to go but I can't switch to using that until this is merged. Once this is merged, I'll configure Bitrise to start failing on new violations.



---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
